src/de.rs

```diff
@@ -0,0 +1,166 @@
+use std::collections::hash_map::Iter;
+
+use hrana_client_proto::Value;
+use serde::{
+    de::{value::SeqDeserializer, IntoDeserializer, MapAccess, Visitor},
+    Deserialize, Deserializer,
+};
+
+use crate::Row;
+
+/// Deserialize from a [`Row`] into any type `T` that implements `Deserialize`.
+pub fn from_row<'de, T: Deserialize<'de>>(row: &'de Row) -> anyhow::Result<T> {
+    let de = De { row };
+    T::deserialize(de).map_err(Into::into)
+}
+
+struct De<'de> {
+    row: &'de Row,
+}
+
+impl<'de> Deserializer<'de> for De<'de> {
+    type Error = serde::de::value::Error;
+
+    fn deserialize_any<V>(self, _visitor: V) -> Result<V::Value, Self::Error>
+    where
+        V: Visitor<'de>,
+    {
+        // visitor.visit_map(RowV { row: &self.row })
```

Comment:
assuming this is why the PR is still draft...

Reply:
Yes, this is why but also we need to add proper error messages. This is a part I have not looked into yet but I assume places like this is where we would error.